source/common/http/filter/buffer_filter.cc

```diff
@@ -38,13 +40,12 @@ FilterDataStatus BufferFilter::decodeData(Buffer::Instance&, bool end_stream) {
     return FilterDataStatus::Continue;
   } else if (callbacks_->decodingBuffer() &&
              callbacks_->decodingBuffer()->length() > config_->max_request_bytes_) {
-    // TODO(htuch): Switch this to Utility::sendLocalReply().
```

Comment:
Can buffer filter now call setDecoderBufferLimit() and have the 413 be by connection manager? On this topic, IMO we should move the rq_too_large_ stat into connection manager (rename if you want). We should have stats in connection manager for sending 413 due to buffering as well as 500 due to buffering, etc.

Reply:
We totally could - I left it as-is to preserve the existing stats.  Will move in my next update if you don't mind the stats going away

Reply:
I think we can lose the existing stats as long as we get new stats in connection manager. :)

Reply:
Ok, one more stats question.  downstream_rq_too_large totally makes sense in the connection manager.  upstream_rs_too_large not so much.  I'm not convinced it's worth the plumbing to make the router know the reason it was torn down was due to buffer limits.  I could just put rs_too_large in the connection manager stats.  WDYT?

Reply:
I think that's fine.